src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/WcfTestDiscoverer.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Infrastructure.Common
+{
+    // Internal helper class for code common to conditional test discovery through
+    // [WcfFact] and [WcfTheory].  It reflects back onto the test method's attributes
+    // to determine whether the test should be skipped.
+    internal class WcfTestDiscoverer
+    {
+
+        internal static IEnumerable<IXunitTestCase> Discover(
+                                                        ITestFrameworkDiscoveryOptions discoveryOptions,
+                                                        IMessageSink diagnosticMessageSink,
+                                                        ITestMethod testMethod,
+                                                        IEnumerable<IXunitTestCase> testCases,
+                                                        IEnumerable<string> conditionMemberNames)
+        {
+
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+ 
+            // Evaluate any [Issue] attributes on this method.
+            // We do this first so that tests we know won't be run will avoid calling
+            // their ConditionalFact conditions (below) which could do unnecessary work
+            // and affect other tests that will be run.
+            IssueAttribute[] issues = testMethodInfo.GetCustomAttributes<IssueAttribute>().ToArray();
+            if (issues.Length > 0)
+            {
+                List<string> issueSkipList = new List<string>();
+                foreach (IssueAttribute issue in issues)
+                {
+                    string skipReason = issue.GetSkipReason(testMethod);
+                    if (skipReason != null)
+                    {
+                        issueSkipList.Add(skipReason);
+                    }
+                }
+
+                if (issueSkipList.Count > 0)
+                {
+                    string skippedReason = string.Format("Active issue(s): {0}", string.Join(",", issueSkipList));
```

Comment:
Good point.  I've restructured a bit so the skip reasons only include the details and let the outer Format add the part that you observed was redundant.  We can tune the message if we find it confusing in practice.